src/java.base/share/classes/java/lang/template/StringTemplate.java

```diff
@@ -0,0 +1,351 @@
+     *
+     * @throws NullPointerException if either @link StringTemplate} is null
+     *
+     * @implSpec Equality is determined by testing equality of the fragments and
+     * the values.
+     */
+    public static boolean equals(Object a, Object b) {
+        Objects.requireNonNull(a, "StringTemplate a should not be null");
+        Objects.requireNonNull(b, "StringTemplate b should not be null");
+        return a instanceof StringTemplate aST && b instanceof StringTemplate bST &&
+                Objects.equals(aST.fragments(), bST.fragments()) &&
+                Objects.equals(aST.values(), bST.values());
+    }
+
+    /**
+     * Returns a StringTemplate composed from a string.
+     *
+     * @param string  single string fragment
+     *
+     * @return StringTemplate composed from string
+     *
+     * @throws NullPointerException if string is null
+     */
+    public static StringTemplate of(String string) {
+        Objects.requireNonNull(string, "string must not be null");
+        return new TemplateRuntime.SimpleStringTemplate(List.of(string), List.of());
+    }
+
+    /**
+     * Returns a StringTemplate composed from fragments and values.
+     *
+     * @implSpec The {@code fragments} list size must be one more that the
+     * {@code values} list size.
+     *
+     * @param fragments list of string fragments
+     * @param values    list of expression values
+     *
+     * @return StringTemplate composed from string
+     *
+     * @throws IllegalArgumentException if fragments list size is not one more
+     *         than values list size
+     * @throws NullPointerException if fragments is null or values is null or if any fragment is null.
+     *
+     * @implNote Contents of both lists are copied to construct immutable lists.
+     */
+    public static StringTemplate of(List<String> fragments, List<Object> values) {
+        Objects.requireNonNull(fragments, "fragments must not be null");
+        Objects.requireNonNull(values, "values must not be null");
+        if (values.size() + 1 != fragments.size()) {
+            throw new IllegalArgumentException(
+                    "fragments list size is not one more than values list size");
+        }
+        fragments = List.copyOf(fragments);
+        values = TemplateRuntime.toList(values.toArray().clone());
+        return new TemplateRuntime.SimpleStringTemplate(fragments, values);
+    }
+
+    /**
+     * Creates a string that interleaves the elements of values between the
+     * elements of fragments.
+     *
+     * @param fragments  list of String fragments
+     * @param values     list of expression values
+     *
+     * @return String interpolation of fragments and values
+     *
+     * @throws NullPointerException fragments or values is null or if any of the fragments is null
+     */
+    public static String interpolate(List<String> fragments, List<Object> values) {
```

Comment:
Should be `String interpolate(List<String> fragments, List<?> values) {` as above

Reply:
Changing

Reply:
Have to backtrack short term. This change forces StringTemplate.values() to return List<?>. Will have to think this through (affects the JLS et al)

Reply:
Worked this through for both StringTemplate.of and StringTemplate.interpolate